x/gov/endblocker.go

```diff
@@ -53,13 +53,31 @@ func EndBlocker(ctx sdk.Context, keeper Keeper) sdk.Tags {
 		passes, tallyResults := tally(ctx, keeper, activeProposal)
 
 		var tagValue string
+		var logmsg string
 		if passes {
 			keeper.RefundDeposits(ctx, activeProposal.ProposalID)
 			activeProposal.Status = StatusPassed
-			tagValue = tags.ActionProposalPassed
+
+			handler := keeper.router.GetRoute(activeProposal.ProposalRoute())
+
+			cctx, write := ctx.CacheContext()
+
+			// handler is state mutating logic depending on the proposal content.
+			// handler may mutate the state or not
+			// if handler failes no state mutation happened and the err msg is logged
+			contentErr := handler(cctx, activeProposal.Content)
+			if contentErr == nil {
+				tagValue = tags.ActionProposalPassed
+				logmsg = "passed"
+				write()
```

Comment:
What's this? Maybe the function can be more descriptive?

Reply:
I will add comment - it is the cache writer which applies the writing over the caehed(by `CacheContext`) multistore onto the underlying.